lib/compiler_rt/x86_cpu_model.zig

```diff
@@ -0,0 +1,281 @@
+        inline for (@typeInfo(Target.x86.Feature).Enum.fields) |f| {
+            const index: Target.Cpu.Feature.Set.Index = f.value;
+            if (comptime !@hasField(Feature, f.name)) continue;
+            if (detected.isEnabled(index)) setFeature(&features, @field(Feature, f.name));
+        }
+
+        break :blk features;
+    };
+
+    cpu.vendor = switch (vendor) {
+        .intel => blk: {
+            cpu.type, cpu.subtype = getIntelTypeAndSubtype(family, model, features);
+            break :blk .intel;
+        },
+        .amd => blk: {
+            cpu.type, cpu.subtype = getAmdTypeAndSubtype(family, model, features);
+            break :blk .amd;
+        },
+        else => .other,
+    };
+    cpu.features[0] = features[0];
+    cpu_extra_features = features[1..].*;
+
+    return 0;
+}
+
+const Model = extern struct {
+    vendor: Vendor = .unknown,
+    type: Type = .unknown,
+    subtype: Subtype = .unknown,
+    features: [1]u32 = .{0},
+};
+
+const FeatureSet = [feature_set_size]u32;
+comptime {
+    inline for (@typeInfo(Feature).Enum.fields) |f| {
+        if (f.value >= @bitSizeOf(FeatureSet))
+            @compileError(@import("std").fmt.comptimePrint("Feature.{s} ({}) bitindex too large", .{ f.name, f.value }));
+    }
+}
+
+fn hasFeature(set: FeatureSet, f: Feature) bool {
+    const f_value = @intFromEnum(f);
+    const bit = set[f_value / 32] & (@as(u32, 1) << @as(u5, @intCast(f_value % 32)));
+    return bit != 0;
+}
+fn setFeature(set: *FeatureSet, f: Feature) void {
+    const f_value = @intFromEnum(f);
+    set[f_value / 32] |= (@as(u32, 1) << @as(u5, @intCast(f_value % 32)));
+}
+
+const Vendor = @import("cpu_model/x86.zig").Vendor;
+const Type = @import("cpu_model/x86.zig").Type;
+const Subtype = @import("cpu_model/x86.zig").Subtype;
+const Feature = @import("cpu_model/x86.zig").Feature;
+
+const feature_set_size = blk: {
+    var max_index: comptime_int = 0;
+    inline for (@typeInfo(Feature).Enum.fields) |f| {
+        if (f.value > max_index) max_index = f.value;
+    }
+    break :blk (max_index + 32) / 32;
+};
+
+comptime {
+    // Sanity check that every name in Feature has a corresponding name in std.Target.x86.Feature.
+
+    const ignored = std.ComptimeStringMap(void, .{
+        // These are not present in std/target/x86.zig because LLVM does
+        // not yet define them in llvm/lib/Target/X86/X86.td.
+        .{ "avx5124vnniw", {} },
+        .{ "avx5124fmaps", {} },
+    });
+
+    var missing = false;
+    inline for (@typeInfo(Feature).Enum.fields) |field| {
+        if (ignored.has(field.name)) {
+            continue;
+        } else if (!@hasField(Target.x86.Feature, field.name)) {
+            missing = true;
+            @compileLog(field);
+        }
+    }
+    if (missing) @compileError("missing enum values in std.Target.x86.Feature (see compile log)");
+}
+
+// This function should be reviewed after any update to the autogenerated
```

Comment:
Better make this dry and note the functions to review in the file level comment-